packages/controllers/src/plugins/PluginController.ts

```diff
@@ -53,38 +53,56 @@ type RemoveAllPermissionsFunction = (pluginIds: string[]) => void;
 type CloseAllConnectionsFunction = (domain: string) => void;
 type RequestPermissionsFunction = (
   domain: string,
-  requestedPermissions: IRequestedPermissions,
-) => IOcapLdCapability[];
+  requestedPermissions: RequestedPluginPermissions,
+) => Promise<IOcapLdCapability[]>;
 type HasPermissionFunction = (
   domain: string,
   permissionName: string,
 ) => boolean;
 type GetPermissionsFunction = (domain: string) => IOcapLdCapability[];
+type TerminateWorkerOf = (pluginName: string) => void;
+type Command = (
+  workerId: string,
+  message: JsonRpcRequest<unknown>,
+) => Promise<unknown>;
+type TerminateAll = () => void;
+type CreatePluginWorker = (metadate: PluginWorkerMetadata) => Promise<string>;
+type StartPlugin = (
+  workerId: string,
+  pluginData: PluginData,
+) => Promise<unknown>;
 
-interface StoredPlugins {
-  [pluginId: string]: Plugin;
-}
+type StoredPlugins = Record<string, Plugin>;
 
-export interface PluginControllerState {
+export type PluginControllerState = {
+  inlinePluginIsRunning: boolean;
   plugins: StoredPlugins;
-  pluginStates: Record<string, unknown>;
-}
+  pluginStates: {
+    [id: string]: Json;
+  };
+};
 
 export interface PluginControllerMemState {
   inlinePluginIsRunning: boolean;
-  plugins: { [pluginId: string]: SerializablePlugin };
-  pluginStates: Record<string, unknown>;
+  plugins: Record<string, SerializablePlugin>;
+  pluginStates: {
+    [id: string]: Json;
+  };
```

Comment:
As we discussed with @Gudahtt, I also think it makes sense to define a new `StatePropertyMetadata` property to describe the parts of state that should be communicated to the UI. I have a draft PR for it here: https://github.com/MetaMask/controllers/pull/483

Reply:
Oh, also, this is no longer used anywhere. Maybe we should delete it and document the properties that we don't want to serialize.